### DISCUSSION TITLE
Companion App - Incorrect deltas / trend arrows and issues with the time it syncs

### DISCUSSION THREAD

**Discussion by petalina** [NightscoutFoundation/xDrip#3784] [Bug reports]
Hello, since I switched from Dexcom One to Dexcom One+, using the Companion App as source, I have some issues with the arrows / difference of values and the time of update. I think somehow it is stuck in the time the older sensor used to update the values or something (eg some of the time points below - 1st pic: 8.47 / 103 mg/dl - 8.51 / 103 mg/dl - 8.56 / 103 mg/dl - 8.57 / 113 mg/dl, but delta: +41 mg/dl instead of 10 mg/dl) but I can't understand why the deltas it displays are so large even though the diagram and the values are correct. Also, at the same time the difference may be correct in the widget but with the wrong arrow and when I open the app both the arrow and difference are wrong. 
I installed the latest version of xDrip+ yesterday and also tried to Restart Collector and restart my phone.

![Screenshot 2024-11-26 162155](https://github.com/user-attachments/assets/f2d94112-a6f4-4ca9-bd7e-a27cec0d57fb)
![Screenshot 2024-11-26 163223](https://github.com/user-attachments/assets/678a3092-1cc5-4e76-ad6f-90d6dd4a3e99)

**Comment by Navid200** [comment]
Thanks for reporting this.
I would like to figure out why the delta value is incorrect.
I don't expect the delta or arrow to represent the delta or the arrow shown by the companion app.
But, what you are showing is a case where the delta xDrip is showing does not agree with the delta between the current and previous value.

The following is not relevant.  However, for transparency, I always include it.  You don't need to use the companion app if you don't need Clarity: https://navid200.github.io/xDrip/docs/Dexcom/Clarity.html  

Regardless of that that, we need to figure out why this is happening with the xDrip delta.

**Reply by petalina** [reply]
Thank you.
I need to use Clarity even though I wish I didn't.

Is there any chance that it uses the +10 md/dl delta in 1 minute (according to the time points but not the reality / Dexcom app) as 10 mg/dl/min and converts it to an "expected" +41 mg/dl in 4 extra minutes? 

Is there any other way to try to make it sync at the appropriate time?

**Reply by Navid200** [reply]
Please click on "Create issue from discussion" in the right pane.
